### PR TITLE
County labels and painted layers toggle

### PR DESCRIPTION
**Pull request by raphaellaude**
## Description

- Splits sidebar into tabs following existing Districtr layout
- Adds legend (and utils) to toggle visibility of county layers and block layers
- Move basemap layers back to a ts file so that next will build without type errors. I think it loads faster anyway

> [!CAUTION]
> The PaintByCounty toggle doesn't actually make you paint by county for now. That's a slightly larger feature that should have it's own PR.

## Reviewers

- Primary: @mariogiampieri 
- Secondary: @bailliekova 

## Checklist

- [x] Added/Updated related documentation (if applicable).
- [ ] Added/Updated related unit tests (if applicable).

## Screenshots (if applicable):

https://github.com/user-attachments/assets/a490bc12-05d8-4b10-8461-de81c496cd87

### CODE REVIEW

**Comment by raphaellaude** [comment]
Will want to rebase branch once #71 is merged

File: app/src/app/store/mapStore.ts
Comment (mariogiampieri) [inline app/src/app/store/mapStore.ts:80]: I'm not sure this is the issue exactly but I worry that there are collisions with the map state if changes are also coming from the zustand store. We should be using `subscribe`s for this: https://github.com/pmndrs/zustand?tab=readme-ov-file#using-subscribe-with-selector

File: app/src/app/components/sidebar/PaintByCounty.tsx
Comment (mariogiampieri) [inline app/src/app/components/sidebar/PaintByCounty.tsx:19]: I left some other comments in `mapStore.ts`, but I get an error where there's some rendering confusion between here and the `Map` object: 
image
I put a full stack trace at 140 in `mapStore.ts`

Comment (mariogiampieri) [reply app/src/app/components/sidebar/PaintByCounty.tsx:19]: It's worth investigating a bit more but I think we will also want to factor out some `useEffects` in `Map` regardless, but there may be a re-render happening there when I toggle this

File: app/src/app/constants/configuration.ts
Comment (mariogiampieri) [inline app/src/app/constants/configuration.ts:1917]: I take it the above is the same as #71 but you were waiting for my review- apologies if that's so!

File: app/src/app/constants/configuration.ts
Comment (mariogiampieri) [inline app/src/app/constants/configuration.ts:1928]: if this isn't passed in as a JSON, I think we'd want to pull out the file names as constants too instead of being hardcoded in

File: app/src/app/store/mapStore.ts
Comment (mariogiampieri) [inline app/src/app/store/mapStore.ts:13]: I added some notes but I don't think we should be augmenting the map from the store- we can `subscribe` to changes and update that way but shouldn't call non-store functions inside

File: app/src/app/store/mapStore.ts
Comment (mariogiampieri) [inline app/src/app/store/mapStore.ts:81]: I also don't love adding in these hard coded layer ids like this- shouldn't it follow the `layer` argument that's passed in somehow?

**Review by mariogiampieri — CHANGES_REQUESTED** [review]
Per some of the notes, I was seeing a state collision issue in `PaintByCounty.tsx`.

File: app/src/app/store/mapStore.ts
Comment (raphaellaude) [reply app/src/app/store/mapStore.ts:140]: Yeah I was wondering about that. Only happens the first time

File: app/src/app/components/sidebar/Layers.tsx
Comment (raphaellaude) [reply app/src/app/components/sidebar/Layers.tsx:45]: Do you mean the numbering of districts or the checkbox values?

File: app/src/app/store/mapStore.ts
Comment (raphaellaude) [reply app/src/app/store/mapStore.ts]: Needed to use a useEffect!

File: app/src/app/store/mapStore.ts
Comment (raphaellaude) [reply app/src/app/store/mapStore.ts]: Makes sense and this is much nicer.

File: app/src/app/components/sidebar/PaintByCounty.tsx
Comment (raphaellaude) [reply app/src/app/components/sidebar/PaintByCounty.tsx:18]: Just was a bad implementation in PaintByCounty. I was attempting to modify state within a component and outside a useEffect 🙃

File: app/src/app/constants/configuration.ts
Comment (raphaellaude) [reply app/src/app/constants/configuration.ts]: Fair enough – for now if it's okay I'm going to leave it in since counties don't really change very often

File: app/src/app/store/mapStore.ts
Comment (raphaellaude) [reply app/src/app/store/mapStore.ts:14]: Makes total sense and the subscribe approach is much nicer

File: app/src/app/store/mapStore.ts
Comment (raphaellaude) [reply app/src/app/store/mapStore.ts]: I think it's okay for now since this is for the special block ids. Refactored into a subscribe

**Review by mariogiampieri — APPROVED** [review]
lgtm, nice work!

**Pull request merged**
